### PR TITLE
Separate parsing in two passes

### PR DESCRIPTION
**Pull request by MightyCreak**
Changes: 

- [x] Separate parsing the text files and merging the results (2567a0b)
- [x] Ensure the resulting XML is at least better than the current result
- [x] Clean more code (remove `addExtension2` and `addSubExtension2`) (16c1efb)
- [x] Parse only needed commits (2cc675d)

### CODE REVIEW

**Comment by MightyCreak** [comment]
@Xenopathic Yop! I made quite an important PR, it simplifies the understanding of the parsing by separating it into two passes:

1. Parse each commit and write the result in an XML file (in `private/commits/`)
2. Parse all the XML file of the commits and merge them to generate `gl3.xml`

____________________________________________________________

I had quite some trouble to understand how the commits where merged together. Eventually I found the problem and it was that the methods adding new extensions or new supported drivers was expecting not entirely parsed parameters (for instance, the name of the driver was actually its name followed by the hint). The constructors of these classes was actually finishing the job.

In order to have a simpler diff, I preferred not to change these constructors. Instead I decided to create these classes outside of the adding methods. This way I could control when we should finish the parsing during the construction or not. So that's why you'll find some `new OglExtension` directly in `OglParser`.
____________________________________________________________

In the end I verified the end result (e.g. `gl3.xml`) with the previous one and it seems that it also fixes #66. Now the hint `llvmpipe and softpipe have fake Multisample anti-aliasing support` is properly attributed to me (huhu :wink:) and with it, the right date. You can see that, right now, the hint date of [the extension](http://mesamatrix.net/#Version_OpenGL3.0-GLSL1.30_Extension_Multisample_antialiasing) is wrong (2015-07-25).
____________________________________________________________

There are other things to do, but I think it's a good base before going further (like fixing #68).

So, what do you think?

File: lib/Parser/OglSupportedDriver.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
 class OglSupportedDriver
 {
     public function __construct($name, $hints) {
-        $this->setName("<undefined>");
+        $this->setName("%undefined%");
```
Comment (MightyCreak) [inline lib/Parser/OglSupportedDriver.php]: This is because it creates conflicts with XML tags if the driver isn't found.

Comment (Xenopathic) [inline lib/Parser/OglSupportedDriver.php]: Can we not leave this as `null`, to save the hassle?

**Comment by Xenopathic** [comment]
Code looks good.

File: lib/Parser/OglSupportedDriver.php (hunk 1 above)
Comment (MightyCreak) [inline lib/Parser/OglSupportedDriver.php]: True, I'll fix that

**Pull request merged**